src/webui/api/synccontroller.cpp

```diff
@@ -847,12 +847,15 @@ void SyncController::torrentPeersAction()
 
     for (const BitTorrent::PeerInfo &pi : peersList)
     {
-        if (pi.address().ip.isNull()) continue;
+        const bool useI2PSocket = pi.useI2PSocket();
+        if (pi.address().ip.isNull() && !useI2PSocket) continue;
+
+        const QString address = useI2PSocket ? pi.I2PAddress() : pi.address().ip.toString();
 
         QVariantMap peer =
         {
-            {KEY_PEER_IP, pi.address().ip.toString()},
-            {KEY_PEER_PORT, pi.address().port},
+            {KEY_PEER_IP, address},
```

Comment:
Would it be better to add another field for I2P address?
```c++
const QString KEY_PEER_I2P = u"i2p"_s;
```
And `KEY_PEER_IP` would be empty in case of an I2P address.

Reply:
> Would it be better to add another field for I2P address?

:+1:


> And `KEY_PEER_IP` would be empty in case of an I2P address.

IMO, it should be omitted at all (and KEY_PEER_PORT).